torch/csrc/autograd/compiled_autograd.h

```diff
@@ -0,0 +1,585 @@
+#pragma once
+#include <c10/core/impl/TorchDispatchModeTLS.h>
+#include <torch/csrc/autograd/engine.h>
+#include <torch/csrc/utils/python_stub.h>
+#include <torch/csrc/utils/torch_dispatch_mode.h>
+#include <iostream>
+#include <typeindex>
+#include <vector>
+
+namespace torch {
+namespace autograd {
+namespace generated {
+struct TypeAndSize;
+}
+
+struct SizeInput {
+  enum DynType : uint8_t { STATIC = 0, DYNAMIC = 1 };
+  DynType dyn_type;
+  int64_t value;
```

Comment:
nit: uninitialized struct members
hypernit: Do we follow convention of prefixing struct member with the struct name?